### ISSUE TITLE
Functions passed as parameters trigger no-await-sync-query

### ISSUE THREAD

**Issue by jeblackburn** [testing-library/eslint-plugin-testing-library#276]
Hello,

I have an async function that takes a query function as a parameter:

```
export const chooseElementFromSomewhere = async (
    text: string, 
    getAllByLabelText: (text: string | RegExp, options?: SelectorMatcherOptions) => HTMLElement[],
): Promise<void> => {
  const someElement = getAllByLabelText(text)[0].parentElement;
  ...
  await someOtherAsyncFunction();
};

```
(somewhere in my test code)
`await chooseElementFromSomewhere('someTextToUseInAQuery', getAllByLabelText);`

Calls to this function are triggering this eslint rule:
`'getAllByLabelText' does not need 'await' operator  testing-library/no-await-sync-query`

I think it's a bug in the eslint parsing/interpreting rule. The expected behavior would be for eslint to pass my code as it is.

**Comment by Belco90** [comment]
Hi @jeblackburn! Thanks for opening this issue. It is a bug in our plugin indeed.

The problem is that there is a call expression `chooseElementFromSomewhere()` with a Testing Library query identifier inside `getAllByLabelText` but the latter is not part of the callee but the arguments. We have a mechanism in future v4 to detect this (which already fixed several cases like this across the plugin). I'll see how easy it's to fix in current v3 and let you know about it, and make sure this is fixed in v4 too.

**Comment by Belco90** [comment]
Confirmed: this will be fixed on v4. Thank you for reporting and sorry for not being able to fix it on current v3.
